### ISSUE TITLE
set cookie/h11 problem on windows10

### ISSUE THREAD

**Issue by manatlan** [Kludex/starlette#188]
h11(0.8.1), latest starlette/guvicorn ... on python3.7.0/windows10

this simple app:
```python
from starlette.applications import Starlette
from starlette.responses import RedirectResponse
import uvicorn

app = Starlette()
app.debug=True

@app.route('/')
async def simplest(request):
    r=RedirectResponse( "http://example.com" )
    r.set_cookie("mycake", "myvalue", path="/", max_age=546546546, httponly=True)
    return r

if __name__=="__main__":
    uvicorn.run(app, host="0.0.0.0", port=8080)
```

produce

```
  File "...\h11\_util.py", line 96, in validate
    raise LocalProtocolError(msg)
h11._util.LocalProtocolError: Illegal header value b' mycake=myvalue; HttpOnly; Max-Age=546546546; Path=/'
```

**Comment by Witekkq** [comment]
Looks that header should not contain space at the begining and at the end, looks that httptools on unix system allows that and h11 does not.

Based on [https://tools.ietf.org/html/rfc2616](url)  leading space is not allowed (HTTP/1.1)

Can I create PR with this change?

**Comment by lovelydinosaur** [comment]
@Witekkq please do, yes!

**Comment by Witekkq** [comment]
#194 not sure if that's the best solution to simply strip trailing spaces from cookie value, but no other idea.

**Comment by lovelydinosaur** [comment]
Closed via #194
